traildb/traildb.py

```diff
@@ -275,50 +346,82 @@ def __init__(self, path):
         self.num_events = lib.tdb_num_events(db)
         self.num_fields = lib.tdb_num_fields(db)
         self.fields = [lib.tdb_get_field_name(db, i) for i in xrange(self.num_fields)]
-        self._event_cls = namedtuple('event', self.fields, rename=True)
+        self._event_cls = mk_event_class(self.fields, self.get_item_value)
         self._uint64_ptr = pointer(c_uint64())
 
     def __del__(self):
         if hasattr(self, '_db'):
             lib.tdb_close(self._db)
 
     def __contains__(self, uuidish):
-        """Return True if UUID or Trail ID exists in this TrailDB."""
+        """:returns: True if UUID or Trail ID exists in this TrailDB."""
         try:
             self[uuidish]
             return True
         except IndexError:
             return False
 
     def __getitem__(self, uuidish):
-        """Return a cursor for the given UUID or Trail ID."""
+        """:returns: a cursor for the given UUID or Trail ID."""
         if isinstance(uuidish, basestring):
             return self.trail(self.get_trail_id(uuidish))
         return self.trail(uuidish)
 
     def __len__(self):
-        """Return the number of trails."""
+        """:returns: The number of trails in the TrailDB."""
         return self.num_trails
 
-    def trails(self, selected_uuids=None, **kwds):
+    def trails(self, selected_uuids=None, distinct_cursors=False, **kwds):
         """
         Iterate over all trails in this TrailDB.
 
-        The selected_uuids keyword argument can be used to select a subset
-        of trails to iterate over. Missing uuids are skipped over.
+        :param selected_uuids: If passed, only go through the UUIDs passed in
+          this argument. It should be an iterable that yields hex UUIDs.
```

Comment:
Should we add an explicit check of each UUID as we iterate over it? (Or do we check already?)  If performance is a concern there, then I guess any bad strings just won't be matched and a trail won't be returned for them.  Maybe we need to add a comment that entries that aren't found are silently skipped and no error is raised then?